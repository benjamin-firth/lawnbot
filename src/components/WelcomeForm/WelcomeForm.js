import React from 'react';
import styled from 'styled-components';

const WelcomeForm = () => {
  return (
    <StyledWelcomeForm>
      <StyledWelcomeTitle>
        <h3>Thanks for filling in your lawn area!</h3>
      </StyledWelcomeTitle>
      <form>
        <label>Your approximate lawn size (sq ft)</label>
        <input placeholder="Example 1,500"/>
      </form>
    </StyledWelcomeForm>
  );
};

const StyledWelcomeForm = styled.section`
  background-color: #fff;
  border-radius: 5px;
  display: flex;
  flex-direction: column;

  form {
    align-items: center;
    display: flex;
    flex-direction: column;
    padding: 25px;
    font-size: 1em;
  }

  input {
    height: 22px;
    margin-top: 10px;
    width: 82%;
  }
`;

const StyledWelcomeTitle = styled.div`
  background-color: #36b459;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  color: #fff;
  padding: 20px;
`;

export default WelcomeForm;