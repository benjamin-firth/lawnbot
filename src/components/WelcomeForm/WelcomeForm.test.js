import React from 'react';
import WelcomeForm from './WelcomeForm';
import { shallow } from 'enzyme';

jest.mock("react-redux", () => ({
  useDispatch: () => jest.fn()
}));

describe('WelcomeForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<WelcomeForm />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});