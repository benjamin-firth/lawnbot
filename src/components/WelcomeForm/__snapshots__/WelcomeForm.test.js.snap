// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeForm should match the snapshot 1`] = `
<styled.section>
  <styled.div>
    <h3>
      Thanks for filling in your lawn area!
    </h3>
  </styled.div>
  <form>
    <label>
      Your approximate lawn size (sq ft)
    </label>
    <input
      id="yardSubmit"
      onChange={[Function]}
      placeholder="Example 1,500"
      type="number"
    />
  </form>
</styled.section>
`;
